benchmarks/dashboard.md

```diff
@@ -0,0 +1,144 @@
+# IREE Performance Dashboard
+
+This documentation explains IREE's performance dashboard (https://perf.iree.dev).
+A [Buildkite pipeline](https://buildkite.com/iree/iree-benchmark) runs on each
+commit to the `main` branch and posts those results to the dashboard.
+
+## Benchmarking philosophy
+
+Benchmarking and interpreting results properly is a delicate task. We can record
+metrics from various parts of a system, but depending on what we are trying to
+evaluate, those numbers may or may not be relevant. For example, for somebody
+working solely on better kernel code generation, the end-to-end model reference
+latency is unlikely meaningful given it also includes runtime overhead. The
+environment could also vary per benchmark run in uncontrollable ways, causing
+instability in the results. This is especially true for Android, where a tight
+compromise between performance and thermal/battery limits is made. Too many
```

Comment:
could rephrase "Android" to "mobile and embedded systems"
(doing any development involving Android is a pain, and I'd like to not paint ourselves further into a corner with infra, docs, etc.)